packages/internals/src/engine-commands/formatSchema.ts

```diff
@@ -1,52 +1,120 @@
+      debug('using a schema string directly')
+      return _schema
+    })
+    .when(isSchemaPathOnly, ({ schemaPath: _schemaPath }) => {
+      debug('reading a schema from a file')
+
+      if (!fs.existsSync(_schemaPath)) {
+        throw new Error(`Schema at ${schemaPath} does not exist.`)
+      }
+
+      const _schema = fs.readFileSync(_schemaPath, { encoding: 'utf8' })
+      return _schema
+    })
+    .exhaustive()
 
-  const options = {
-    env: {
-      RUST_BACKTRACE: process.env.RUST_BACKTRACE ?? '1',
-      ...(showColors ? { CLICOLOR_FORCE: '1' } : {}),
+  const formattedSchema = await handleFormatPanic(
+    () => {
+      // the only possible error here is a Rust panic
+      return formatWASM(schemaContent)
     },
-    maxBuffer: MAX_BUFFER,
-  } as execa.Options
+    { schemaPath, schema } as FormatSchemaParams,
+  )
 
-  let result
+  return formattedSchema
+}
 
-  if (process.env.FORCE_PANIC_PRISMA_FMT) {
-    result = await execa(prismaFmtPath, ['debug-panic'], options)
+function handleFormatPanic<T>(tryCb: () => T, { schemaPath, schema }: FormatSchemaParams) {
+  try {
+    return tryCb()
+  } catch (e: unknown) {
+    const wasmError = e as Error
+    throw new RustPanic(
+      /* message */ wasmError.message,
+      /* rustStack */ wasmError.stack || 'NO_BACKTRACE',
+      /* request */ '@prisma/prisma-fmt-wasm format',
+      ErrorArea.FMT_CLI,
+      schemaPath,
+      schema,
+    )
   }
+}
 
-  if (schemaPath) {
-    if (!fs.existsSync(schemaPath)) {
-      throw new Error(`Schema at ${schemaPath} does not exist.`)
-    }
-    result = await execa(prismaFmtPath, ['format', '-i', schemaPath], options)
-  } else if (schema) {
-    result = await execa(prismaFmtPath, ['format'], {
-      ...options,
-      input: schema,
-    })
+type DocumentUri = string
+
+type TextDocument = {
+  /**
+   * The associated URI for this document. Most documents have the __file__-scheme, indicating that they
+   * represent files on disk. However, some documents may have other schemes indicating that they are not
+   * available on disk.
+   */
+  readonly uri: DocumentUri
+}
+
+// Part of the LSP spec: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#documentFormattingParams
+// These are only the parts we are interested in.
+type DocumentFormattingParams = {
+  textDocument: TextDocument
+  options: {
+    // this is the only property currently considered by Rust, the rest are ignored but are needed for successfully unmarshaling the `DocumentFormattingParams`
+    // and be compatible with the LSP spec.
+    // The WASM formatter may fail silently on unmarshaling errors (a `warn!` macro is used in the Rust code, but that's not propagated to WASM land).
+    tabSize: number
+
+    insertSpaces: boolean
   }
+}
+
+const defaultDocumentFormattingParams: DocumentFormattingParams = {
+  textDocument: { uri: 'file:/dev/null' },
+  options: {
+    tabSize: 2,
```

Comment:
To double-check: the default was 2 tabs before this change?

Reply:
`2 tabs` is what's applied when I use `Format document` in VSCode when formatting a Prisma schema. Now that you ask, I wonder whether that's coming from my IDE's settings or the language-tools repo

Reply:
This uses a `params` argument: https://github.com/prisma/language-tools/blob/9534af3cdabd7b2dd1277f17193d596601f43e4c/packages/language-server/src/server.ts#L199-L200

Reply:
And `prisma format` indeed uses `2` spaces for a tab indendation

Reply:
E.g.:

```prisma
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}
```

Reply:
Yes, but we are not in language-tools here.
So what we could do, is to check what the CLI was doing and what it does now with some schemas locally, for example:
- schema using 2 tabs
- schema using 4 tabs
- schema using 2 spaces
- schema using 4 spaces

So we know if we are changing something here to the output.

Reply:
Because I don't think we test these cases yet

Reply:
Ok, should we actually do this in this PR, or could this wait a later PR?

Reply:
Would be good to do a quick check, we can finish this on Monday

Reply:
Added this customization in [this commit](https://github.com/prisma/prisma/pull/14766/commits/19aead7085e1bf1ec25e355ab7ada1b11b6155d5). Note that, currently, `options.tabSize` is the only formatting option taken into consideration Rust-wise.